incubator/drone/templates/deployment-agent.yaml

```diff
@@ -28,21 +28,34 @@ spec:
         image: "{{ .Values.images.agent.repository }}:{{ .Values.images.agent.tag }}"
         imagePullPolicy: {{ .Values.images.agent.pullPolicy }}
         env:
-          - name: DRONE_SERVER
-            value: {{ template "drone.fullname" . }}:9000
-          - name: DRONE_SECRET
-            valueFrom:
-              secretKeyRef:
-                name: {{ template "drone.fullname" . }}
-                key: secret
-          - name: DOCKER_HOST
-            value: tcp://localhost:2375
-          {{ range $key, $value := .Values.agent.env }}
-          - name: {{ $key }}
-            value: {{ $value | quote }}
-          {{ end }}
+        - name: DRONE_SECRET
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "drone.fullname" . }}
+              key: secret
+        - name: DRONE_SERVER
+          value: {{ template "drone.fullname" . }}:9000
+{{- if .Values.dind.enabled }}
+        - name: DOCKER_HOST
+          value: tcp://localhost:2375
+{{- end }}
+        {{ range $key, $value := .Values.agent.env }}
+        - name: {{ $key }}
+          value: {{ $value | quote }}
+        {{ end }}
```

Comment:
If you keep the original order of the ENV variables it is much clearer which changes you did.

Reply:
Right, except indenting back 1 step all env variables to keep spacing consistent throughout the file, I only added the `if` and `end`.

Reply:
Actually, it's not consistent in `volumeMounts` either. Guess I should reset the indentation as before and maybe open a different PR to fix styling issues.